pkg/controller/pxc/version.go

```diff
@@ -98,26 +104,37 @@ func (r *ReconcilePerconaXtraDBCluster) ensurePXCVersion(cr *api.PerconaXtraDBCl
 		return errors.New("cluster is not ready")
 	}
 
-	new, err := vs.GetExactVersion(cr.Spec.UpgradeOptions.Apply, cr.Status.PXC.Version)
+	newVersion, err := vs.GetExactVersion(versionMeta{
+		Apply:         cr.Spec.UpgradeOptions.Apply,
+		Platform:      string(cr.Spec.Platform),
+		KubeVersion:   string(r.serverVersion.Info.GitVersion),
```

Comment:
no need in conversion, these params are already strings

Reply:
fixed